src/Symfony/Component/Yaml/Inline.php

```diff
@@ -717,8 +721,12 @@ private static function evaluateScalar($scalar, $flags, $references = [])
                         return null;
                     case 0 === strpos($scalar, '!php/const'):
                         if (self::$constantSupport) {
+                            if ((false === $const = substr($scalar, 11)) || '' === $const) {
```

Comment:
i think we should start deprecating/throwing for empty strings (or blanks: https://3v4l.org/EU5Sq) in 5.1 :thinking: not sure it's worth it for 3.4, but i agree we should only accept readable values.

Reply:
I think I threw too fast on 3.4 indeed. The error is actually a notice so someone can technically have a working code with the current behavior. Maybe we should indeed just let 3.4 as it is and start improving it on master.

Reply:
I agree, we should probably make we get the empty string here instead of a PHP error and deprecate in 5.1 instead.

Reply:
`parseScalar()` does not work with an empty string. You mean we should directly return an empty string here?

Reply:
Yes, so that it looks like this:

```php
if (self::$constantSupport) {
    if (!isset($scalar[11])) {
        return '';
    }

    $i = 0;

    if (\defined($const = self::parseScalar(substr($scalar, 11), 0, null, $i, false))) {
        return \constant($const);
    }
}
```

Reply:
Alright that seems like a good compromise on 3.4.